### PR TITLE
bench: add repro test

### PR DESCRIPTION
**Pull request by dberenbaum**
Closes https://github.com/iterative/dvc-bench/issues/468

Edit: the need for this is to help track performance for issues like https://github.com/iterative/dvc/issues/9085

### CODE REVIEW

File: dvc/testing/benchmarks/cli/commands/test_repro.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    dvc.stage.add(
+        name="copy-2",
+        deps=[source_path],
+        outs=["copy-2"],
+        cmd=f"cp -R {source_path} copy-2",
+    )
+    combine_cmds = [
+        "mkdir combine",
+        "cp -R copy-1 combine/1",
+        "cp -R copy-2 combine/2",
```
Comment (efiop) [inline dvc/testing/benchmarks/cli/commands/test_repro.py:19]: Ideally, if we can, this and commands above should be cross-platform. That's why we have our own python scripts in regular tests. Simple python scripts would be great here.

File: dvc/testing/benchmarks/cli/commands/test_repro.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+def test_repro(bench_dvc, tmp_dir, dvc, dataset):
+    dvc.add(dataset)
```
Comment (efiop) [inline dvc/testing/benchmarks/cli/commands/test_repro.py:2]: You should be able to use `make_dataset`, that will just copy things over to cache/workspace without wasting time dvc-adding from scratch.

Btw, you might be able to just use existing project here, similar to how we test exp show and stuff.

Comment (dberenbaum) [reply dvc/testing/benchmarks/cli/commands/test_repro.py:2]: I think I want to keep it artificial for now so it scales to the various datasets in dvc-bench and doesn't vary too much based on things like model training time.

**Comment by skshetry** [comment]
It's going to be hard to write a fair benchmark for `dvc repro`.

I understand the motivation, but what you are testing here is `dvc commit` and that too is a no-op here because all of those were run already with `run_copy()` and is cached (mtimes, md5 hashes and the actual files).

Even if you made all of them `--no-exec` and tried to benchmark `dvc repro`, there is I/O happening in the subprocess `cmd`, which will be a significant part of the runtime.

**Pull request closed without merging**